### PR TITLE
DOC-12385: Copy across missing snippets

### PR DESCRIPTION
**Pull request by ElliotFrancisHunter**
Move snippets from p2psync-websocket.cs to Program.cs

### CODE REVIEW

**Comment by borrrden** [comment]
This has broken compilation and cannot be merged

**Review by borrrden — CHANGES_REQUESTED** [review]
This needs to be written in a way that matches the existing code in Program.cs.  It _cannot_ just be plopped in.  This is what the GH checks are meant to prevent.  These snippets should be able to be compiled.

**Comment by osfameron** [comment]
Thanks @borrrden - who of the devs can we ask to look at that?
Can you let us know which ticket type to create and link to the DOC ticket above?

**Comment by borrrden** [comment]
Ideally it would be the person who wrote them that puts them in correctly.  Otherwise it's going to be me.  Are all of these snippets needed?  Some of them are just one line which seems a bit excessive to create an entire self contained snippet for.

**Review by borrrden — COMMENTED** [review]
Would like some clarification here on whether these are all important.  There are a lot of tags in here including "how to call start" which seems pretty obvious, does it really need a snippet?

Anyway perhaps a description of what each one wants to do is in order because some of them are suspicious.  I'll put in comments inline.

File: modules/csharp/examples/code_snippets/Program.cs [hunk 1]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
     // end::merge-conflict-resolver[]
  }
 
-#warning p2p-act-rep-func used, but contains nothing
 // tag::p2p-act-rep-func[]
+// . . . preceding code. for example . . .
+    private static ListenerToken _thisListenerToken;
+    var Database thisDB;
+    // . . . other code . . .
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2459]: What is this meant to represent?

File: modules/csharp/examples/code_snippets/Program.cs [hunk 2]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
 // Set autopurge option
 // here we override its default
+thisConfig.EnableAutoPurge = false; // <.>
 
 // end::autopurge-override[]
 
-#warning p2p-act-rep-config-cont used, but contains nothing
 // tag::p2p-act-rep-config-cont[]
 // Configure Sync Mode
+thisConfig.Continuous = true; // default value
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2476]: This is not the default for this property

File: modules/csharp/examples/code_snippets/Program.cs [hunk 3]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
+              Console.WriteLine("Replication stopped");
+          }
+        });
 
 // end::p2p-act-rep-add-change-listener[]
 
-#warning p2p-act-rep-start used, but contains nothing
 // tag::p2p-act-rep-start[]
 // Start replicator
+ thisReplicator.Start(); // <.>
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2514]: This is the same as starting any other replicator...does it really need a separate snippet?

File: modules/csharp/examples/code_snippets/Program.cs [hunk 4]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
-#warning p2p-act-rep-config-cacert used, but contains nothing
 // tag::p2p-act-rep-config-cacert[]
 // Configure Server Security -- only accept CA certs
+thisConfig.AcceptOnlySelfSignedServerCertificate = false; // <.>
+
 // end::p2p-act-rep-config-cacert[]
 
-#warning p2p-act-rep-config-cacert-pinned used, but contains nothing
 // tag::p2p-act-rep-config-cacert-pinned[]
 // Only CA Certs accepted
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2527]: Misleading comment here.  Only certificates issued by this CA are accepted, not any CA cert.

File: modules/csharp/examples/code_snippets/Program.cs [hunk 5]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
+
 // end::p2p-act-rep-config-cacert-pinned[]
 
-#warning p2p-act-rep-status used, but contains nothing
 // tag::p2p-act-rep-status[]
+_thisReplicator.Stop();
+    while (_thisReplicator.Status.Activity != ReplicatorActivityLevel.Stopped) {
+        // Database cannot close until replicators are stopped
+        Console.WriteLine($"Waiting for replicator to stop (currently {_thisReplicator.Status.Activity})...");
+        Thread.Sleep(200);
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2544]: This is quite domain specific and unlikely to make a helpful snippet

File: modules/csharp/examples/code_snippets/Program.cs [hunk 6]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
+    }
+_thisDatabase.Close();
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2546]: Why does this need to be here?

File: modules/csharp/examples/code_snippets/Program.cs [hunk 7]
```diff
@@ -2455,80 +2452,105 @@ public Document Resolve(Conflict conflict)
...
+
 // end::p2p-act-rep-status[]
 
-#warning p2p-act-rep-stop used, but contains nothing
 // tag::p2p-act-rep-stop[]
 // Stop replication.
+thisReplicator.Stop(); // <.>
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/Program.cs:2552]: Same as any other replication...

**Comment by ElliotFrancisHunter** [comment]
Closed as underlying issue has been fixed. Will still need to move these snippets over at some point.

**Pull request closed without merging**